R/flywire-skeleton.R

```diff
@@ -364,8 +364,7 @@ py_skeletor <- function(id,
   if(reroot){
     neuron = reroot_hairball(neuron, k.soma.search = k.soma.search, radius.soma.search = radius.soma.search, brain = brain)
   }
-  if(mesh3d|save.obj){
-    if(is.null(mesh)){
+  if(mesh3d|!is.null(save.obj)){
```

Comment:
@alexanderbates: as a general rule it is better to use || or && for this kind of logic in if statements as those operators short circuit i.e. they stop evaluating when the value of the expression is certain. So if you do 

```
if(shortcheck() | longcheck())
```
both are evaluated even if shortcheck() returns TRUE.

Reply:
If you want convincing, here is some code
```
fastcheck <- function(x=T) {
  message("fast")
  x
}

slowcheck <- function(x=T) {
  message("slow")
  x
}

message("|| operator")
if(fastcheck() || slowcheck()) {
  message("if!")
} else {
  message("else!")
}


message("| operator")
if(fastcheck() | slowcheck()) {
  message("if!")
} else {
  message("else!")
}
```
```
|| operator
fast
if!
| operator
fast
slow
if!
```

Reply:
Thank you, I'll also apply this more generally.